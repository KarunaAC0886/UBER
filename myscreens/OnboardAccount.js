import { SafeAreaView, Text,Pressable, View, StatusBar } from 'react-native'
import React from 'react';
import styles from './styles'


const OnboardAccount = (props) => {
   
  return (
    <SafeAreaView style={styles.onb_acc_mncntr}>

     <Pressable onPress={()=>{props.navigation.navigate('onboardlogin')}}>
        <View style={styles.onb_acc_uber_cntr}>
            <Text style={styles.onb_acc_uber_lbl}>
                Uber
            </Text>
        </View>
        </Pressable>
       <StatusBar/>
    </SafeAreaView>
  )
}
export default OnboardAccount
