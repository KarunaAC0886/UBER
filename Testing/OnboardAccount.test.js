import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { useNavigation } from '@react-navigation/native';
import OnboardAccount from '../myscreens/OnboardAccount';

jest.mock('@expo/vector-icons', () => ({
  FontAwesome: 'FontAwesome', 
}));


jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(),
}));

jest.mock('react-native-responsive-screen', () => ({
  widthPercentageToDP: jest.fn(),
  heightPercentageToDP: jest.fn(),
}));

describe('Account component', () => {
  it('renders correctly', () => {

    useNavigation.mockReturnValue({ navigate: jest.fn() });
    const tree = renderer.create(<OnboardAccount/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
