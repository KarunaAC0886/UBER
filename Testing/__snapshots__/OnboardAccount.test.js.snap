// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account component renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#1976D2",
      "height": undefined,
      "paddingTop": undefined,
      "width": undefined,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderRadius": 30,
          "borderWidth": 0,
          "elevation": 5,
          "height": undefined,
          "justifyContent": "center",
          "width": undefined,
        }
      }
    >
      <Text
        style={
          {
            "color": "#EDF6FF",
            "fontSize": 64,
          }
        }
      >
        Uber
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
